src/components/v2/V2Create/forms/FundingForm/index.tsx

```diff
@@ -116,108 +149,159 @@ export default function FundingForm({ onFinish }: { onFinish: VoidFunction }) {
       dispatch(
         editingV2ProjectActions.setPayoutSplits(splits.map(sanitizeSplit)),
       )
-      dispatch(editingV2ProjectActions.setDuration(fields.duration ?? '0'))
+      dispatch(editingV2ProjectActions.setDuration(durationInSeconds ?? '0'))
 
       onFinish?.()
     },
     [splits, contracts, dispatch, target, targetCurrency, onFinish],
   )
 
-  const onFundingTypeSelect = (newFundingType: FundingType) => {
-    // reset fields if "no target" is selected
-    if (newFundingType === 'no_target') {
-      setTarget(undefined)
-      fundingForm.setFieldsValue({ duration: '0' })
-    }
-    setFundingType(newFundingType)
-  }
-
   // initially fill form with any existing redux state
-  useEffect(() => {
+  useLayoutEffect(() => {
     resetProjectForm()
   }, [resetProjectForm])
 
-  const isFundingTargetSectionVisible =
-    fundingType === 'target' || fundingType === 'recurring'
-  const isFundingDurationVisible = fundingType === 'recurring'
-
+  const onTargetTypeSelect = (type: TargetType) => {
+    setTargetType(type)
+    switch (type) {
+      case 'infinite':
+        setTarget(undefined)
+        break
+      case 'none':
+        setSplits([])
+        setTarget('0')
+        break
+      case 'specific':
+        setTarget('10')
```

Comment:
really really don't think we should default to anything from a ux perspective.

Reply:
Yeah good call